src/Stack/Config/Nix.hs

```diff
@@ -58,21 +58,17 @@ nixCompiler compilerVersion =
       case T.split (== '.') (fromString $ versionString version) of
         x : y : minor ->
           Right $
-          case minor of
-            [] ->
-              -- The minor version is not specified. Select the latest minor
-              -- version in Nixpkgs corresponding to the requested major
-              -- version.
-              let major = T.concat [x, y] in
-              "(let compilers = builtins.filter \
-              \(name: builtins.match \
-              \\"ghc" <> major <> "[[:digit:]]*\" name != null) \
-              \(lib.attrNames haskell.compiler); in \
-              \if compilers == [] \
-              \then abort \"No compiler found for GHC "
-              <> T.pack (versionString version) <> "\"\
-              \else haskell.compiler.${builtins.head compilers})"
-            _ -> "haskell.compiler.ghc" <> T.concat (x : y : minor)
+          -- Select the latest version in Nixpkgs corresponding to the requested
+          -- version.
```

Comment:
I think your change renders this comment invalid, because the minor version is now fixed, not selected.

For example, for 7.10.3 this change no longer gives the flexibility to select `7103`, `7104` and so on, only `7103`.

Why do you change this if your goal is to only accept an additional `Binary` suffix?

Reply:
No, this is still relevant. You can call this with `"7.10"`, which leaves the `minor` list empty, and makes it possible to choose the newest. My intent was to make it so that it would find the `Binary`-suffixed case regardless of whether the minor number was provided, so it was no longer possible to handle the minor case with an exact match.